DataFormats/PatCandidates/interface/PackedCandidate.h

```diff
@@ -786,7 +797,9 @@ namespace pat {
     }
     /// Return true if a bestTrack can be extracted from this Candidate
     bool hasTrackDetails() const { return (packedHits_ != 0 || packedLayers_ != 0); }
-
+    /// Return true if the original candidate had a track associated
+    /// even if the PackedCandidate has no track
+    bool fromTrackCandidate() const { return (packedDz_ != 0 || packedDxy_ != 0); }
```

Comment:
I was scanning some file and see that `! hasTrackDetails` quite frequently has `packedDz_ == 0` but `packedDxy_ == 32768`, which maps to `-0.f`

```suggestion
    bool fromTrackCandidate() const { return (packedDz_ != 0 || (packedDxy_ != 0 && packedDxy_ != 32768) ); }
```